src/Adapter/Product/Image/CommandHandler/DeleteProductImageHandler.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Image\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\Image\Repository\ProductImageRepository;
+use PrestaShop\PrestaShop\Adapter\Product\Image\Uploader\ProductImageUploader;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\Command\DeleteProductImageCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\CommandHandler\DeleteProductImageHandlerInterface;
+
+/**
+ * Handles @see DeleteProductImageCommand
+ */
+class DeleteProductImageHandler implements DeleteProductImageHandlerInterface
+{
+    /**
+     * @var ProductImageUploader
+     */
+    private $productImageUploader;
+
+    /**
+     * @var ProductImageRepository
+     */
+    private $productImageRepository;
+
+    public function __construct(
+        ProductImageUploader $productImageUploader,
+        ProductImageRepository $productImageRepository
+    ) {
+        $this->productImageUploader = $productImageUploader;
+        $this->productImageRepository = $productImageRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function handle(DeleteProductImageCommand $command): void
+    {
+        $image = $this->productImageRepository->get($command->getImageId());
+
+        $this->productImageUploader->remove($image);
+        $this->productImageRepository->delete($image);
```

Comment:
I just wonder do we need to wrap this up?
(I did somehting similar here https://github.com/PrestaShop/PrestaShop/pull/23363/files#diff-6fb575f429e4ed66a563bc17d28fa144a455f80e8c46d36602a0808b6ce228b4R58) 
But Im not sure its worth it. Maybe I did it just because it was convenient, as the VirtualProductUpdater was already introduced. Wdyt? (anyways we should stick to one pattern I suppose, whichever it is)

Reply:
Yes I think it's doable, but I don't have `ProductImageUpdater` in this PR yet I'm implementing the image update where I introduced this updater service

I suggest to move this into the updater in the next PR

Reply:
ok for me